### PR TITLE
Response time test

### PR DESCRIPTION
**Pull request by MaksTkachuk**

### CODE REVIEW

File: api_tests/api_response_time_test.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+def run():
+    logging.info('PYTHON_ENV: %s', PYTHON_ENV)
+    logging.info('Starting time response test...')
+    (elapsed_time, errors) = get_response_time()
+    errors_str = ""
+    is_response_slow = elapsed_time > ACCEPTABLE_RESPONSE_TIME
+    if errors:
+        errors_str = ' '.join(map(str, errors))
+    if config.getboolean('send_discord_notification'):
```
Comment (kalkov) [inline api_tests/api_response_time_test.py:17]:
```python 
if config.getboolean('send_discord_notification') and is_response_slow
```

File: api_tests/api_response_time_test.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+        logging.info('Sending discord notification...')
+        publish_time_response_alert(is_response_slow, errors_str)
```
Comment (kalkov) [inline api_tests/api_response_time_test.py:19]: instead of `errors_str` pass `errors`

File: api_tests/api_response_time_test.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+config = Config(PYTHON_ENV)
+
+def run():
+    logging.info('PYTHON_ENV: %s', PYTHON_ENV)
+    logging.info('Starting time response test...')
+    (elapsed_time, errors) = get_response_time()
+    errors_str = ""
+    is_response_slow = elapsed_time > ACCEPTABLE_RESPONSE_TIME
+    if errors:
```
Comment (kalkov) [inline api_tests/api_response_time_test.py:15]: remove this check

File: api_tests/api_response_time_test.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+from .constants import PYTHON_ENV, NUMBER_OF_RUNS_FOR_TIMING_TEST, ACCEPTABLE_RESPONSE_TIME
+import logging
+
+config = Config(PYTHON_ENV)
+
+def run():
+    logging.info('PYTHON_ENV: %s', PYTHON_ENV)
+    logging.info('Starting time response test...')
+    (elapsed_time, errors) = get_response_time()
+    errors_str = ""
```
Comment (kalkov) [inline api_tests/api_response_time_test.py:13]: remove the assignment

File: api_tests/api_response_time_test.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+    is_response_slow = elapsed_time > ACCEPTABLE_RESPONSE_TIME
+    if errors:
+        errors_str = ' '.join(map(str, errors))
+    if config.getboolean('send_discord_notification'):
+        logging.info('Sending discord notification...')
+        publish_time_response_alert(is_response_slow, errors_str)
+    else:
+        logging.info('Skipping discord notification')
+    logging.info(f"Finished test of {NUMBER_OF_RUNS_FOR_TIMING_TEST} runs, total time {elapsed_time}, errors during the run: {errors_str}")
```
Comment (kalkov) [inline api_tests/api_response_time_test.py:22]: `errors_str` -> `errors`

File: api_tests/discord_bot.py [hunk 6]
```diff
@@ -64,3 +70,15 @@ def build_graphql_error_message(mention, error, triggered_at):
 
 def build_graphql_success_message(triggered_at):
     return f"{triggered_at} GraphQL API check success!"
+
+def build_response_time_error_message(mention, errors_str, triggered_at):
+    report_url = "TBD"
```
Comment (kalkov) [inline api_tests/discord_bot.py:75]: remove report_url

File: api_tests/discord_bot.py [hunk 7]
```diff
@@ -38,6 +38,12 @@ def publish_graphql_alert(error=None):
 
     publish_message(message)
 
+def publish_response_time_alert(is_response_slow, errors_str=None):
```
Comment (kalkov) [inline api_tests/discord_bot.py:41]: pass `errors` list here and in the map it to a string in the function

File: api_tests/discord_bot.py [hunk 8]
```diff
@@ -64,3 +70,15 @@ def build_graphql_error_message(mention, error, triggered_at):
 
 def build_graphql_success_message(triggered_at):
     return f"{triggered_at} GraphQL API check success!"
+
+def build_response_time_error_message(mention, errors_str, triggered_at):
+    report_url = "TBD"
+    return f"""
++++++++++++++++++++++++++++++++++++++++++++++++++
+{mention}
```
Comment (kalkov) [inline api_tests/discord_bot.py:78]: We need to mention several people here, is the code ready for that?

File: api_tests/discord_bot.py [hunk 9]
```diff
@@ -64,3 +70,15 @@ def build_graphql_error_message(mention, error, triggered_at):
...
+API response time is slow!
+Errors encountered: {errors_str}
+Triggered at {triggered_at}
+See report at {report_url}
```
Comment (kalkov) [inline api_tests/discord_bot.py:82]: remove

File: api_tests/discord_bot.py [hunk 10]
```diff
@@ -64,3 +70,15 @@ def build_graphql_error_message(mention, error, triggered_at):
 
 def build_graphql_success_message(triggered_at):
     return f"{triggered_at} GraphQL API check success!"
+
+def build_response_time_error_message(mention, errors_str, triggered_at):
+    report_url = "TBD"
+    return f"""
++++++++++++++++++++++++++++++++++++++++++++++++++
+{mention}
+API response time is slow!
```
Comment (kalkov) [inline api_tests/discord_bot.py:79]: Add the actual response time and the expected one.

File: api_tests/api_helper.py [hunk 11]
```diff
@@ -202,3 +202,24 @@ def get_min_interval(metric):
...
+    errors = []
+    runs = 0
+    started = time.time()
+    while runs < NUMBER_OF_RUNS_FOR_TIMING_TEST:
+        runs += 1
+        try:
+            response = execute_gql(gql_query)
+        except SanError as e:
+            errors.append(e)
+            time.sleep(RETRY_DELAY)
```
[the remark this answers is not in the document]
Comment (MaksTkachuk) [reply api_tests/api_helper.py:223]: yes, talked with Ivan and confirmed that without the delay we can hit the rate limiter here

File: api_tests/api_response_time_test.py [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+    logging.info('PYTHON_ENV: %s', PYTHON_ENV)
+    logging.info('Starting time response test...')
+    (elapsed_time, errors) = get_response_time()
+    is_response_slow = elapsed_time > ACCEPTABLE_RESPONSE_TIME
+    if config.getboolean('send_discord_notification') and is_response_slow:
+        logging.info('Sending discord notification...')
+        publish_time_response_alert(elapsed_time, errors)
+    else:
+        logging.info('Skipping discord notification')
+    logging.info(f"Finished test of {NUMBER_OF_RUNS_FOR_TIMING_TEST} runs, total time {elapsed_time}, errors during the run: {bool(errors)}")
```
Comment (kalkov) [inline api_tests/api_response_time_test.py:45]: We want to see the exact error in the logs, leave: 
```python
{errors}
```
will be outputted as a list.

File: api_tests/discord_bot.py [hunk 13]
```diff
@@ -64,3 +70,14 @@ def build_graphql_error_message(mention, error, triggered_at):
 
 def build_graphql_success_message(triggered_at):
     return f"{triggered_at} GraphQL API check success!"
+
+def build_response_time_error_message(mention, time, errors_str, triggered_at):
+    return f"""
++++++++++++++++++++++++++++++++++++++++++++++++++
+{mention}
+API response time is slow!
+Expected {ACCEPTABLE_RESPONSE_TIME} s, actual {time} s
```
Comment (kalkov) [inline api_tests/discord_bot.py:79]: Expected  -> Acceptable

**Pull request merged**